src/streaming/TextTracks.js

```diff
@@ -442,16 +437,20 @@ function TextTracks() {
 
                 cue.onenter =  function () {
                     if (track.mode === 'showing') {
-                        log('Cue ' + this.startTime + '-' + this.endTime + ' : ' + this.cueHTMLElement.id + ' : ' + this.cueHTMLElement.innerText);
-                        captionContainer.appendChild(this.cueHTMLElement);
-                        scaleCue.call(self, this);
+                        var finalCue = document.createElement('div');
+                        log('Cue enter id:' + this.cueID);
+                        renderHTML(this.isd, finalCue, null, captionContainer.clientWidth, captionContainer.clientHeight);
+                        finalCue.id = this.cueID;
+                        captionContainer.appendChild(finalCue);
```

Comment:
Unfortunately, this doesn't work properly when line padding is to be applied because the parser uses getComputedStyle to calculate the required padding, which only works when the element is attached to the DOM.

So finalCue needs to be appended to captionContainer before renderHTML is called.

See sandflow/imscJS#7